### ISSUE TITLE
Lua errors with Cancelling

### ISSUE THREAD

**Issue by Barov2** [Auctionator/Auctionator#1404]
Please include as much of the following information as possible to help me fix the bug:

## Brief Description of Bug

Lua errors when pressing the button to check the price of items (to possibly cancel a sale). Happens every time.

## Loaded Addons

Auctionator 10.1.10-Wrath.
WoW Classic WotLK, 3.4.2.50250

## Back Trace (LUA Error Output)

```
514x ...tor/Source_Classic/Tabs/DataProviders/Cancelling.lua:249: attempt to compare nil with number
[string "@Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua"]:249: in function `PopulateAuctions'
[string "@Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua"]:120: in function `NoQueryRefresh'
[string "@Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua"]:161: in function `ReceiveEvent'
[string "@Auctionator/Source/Utilities/EventBusMixin.lua"]:77: in function `Fire'
[string "@Auctionator/Source_Classic/Tabs/Cancelling/Mixins/UndercutScan.lua"]:205: in function `ProcessUndercutResult'
[string "@Auctionator/Source_Classic/Tabs/Cancelling/Mixins/UndercutScan.lua"]:181: in function `ReceiveEvent'
[string "@Auctionator/Source/Utilities/EventBusMixin.lua"]:77: in function `Fire'
[string "@Auctionator/Source_Classic/AH/Mixins/Scan.lua"]:94: in function `ProcessSearchResults'
[string "@Auctionator/Source_Classic/AH/Mixins/Scan.lua"]:39: in function <Auctionator/Source_Classic/AH/Mixins/Scan.lua:36>

Locals:
self = Frame {
 0 = <userdata>
 GetEntryAt = <function> defined @Auctionator/Source/Components/DataProviders/DataProvider.lua:100
 ProcessItemString = <function> defined @Auctionator/Source_Classic/Tabs/DataProviders/ItemStringLoading.lua:20
 Sort = <function> defined @Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua:135
 onPreserveScroll = <function> defined @Auctionator/Source/Components/ResultsListing/Mixins/ResultsListing.lua:54
 searchCompleted = false
 OnUpdate = <function> defined @Auctionator/Source/Components/DataProviders/DataProvider.lua:24
 OnShow = <function> defined @Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua:104
 insertedKeys = <table> {
 }
 GetCSV = <function> defined @Auctionator/Source/Components/DataProviders/DataProvider.lua:224
 GetColumnHideStates = <function> defined @Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua:300
 ReceiveEvent = <function> defined @Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua:151
 SetOnEntryProcessedCallback = <function> defined @Auctionator/Source/Components/DataProviders/DataProvider.lua:110
 announcedCompletion = true
 onSearchEnded = <function> defined @Auctionator/Source/Components/ResultsListing/Mixins/ResultsListing.lua:47
 isDirty = true
 IsValidAuction = <function> defined @Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua:169
 processingIndex = 0
 undercutCutoff = <table> {
 }
 results = <table> {
 }
 onResetScroll = <function> defined @Auctionator/Source/Components/ResultsListing/Mixins/ResultsListing.lua:58
 onUpdate = <function> defined @Auctionator/Source/Components/ResultsListing/Mixins/ResultsListing.lua:38
 onSearchStarted = <function> defined @Auctionator/Source/Components/ResultsListing/Mixins/ResultsListing.lua:42
 onEntryProcessed = <function> defined @Auctionator/Source_Classic/Tabs/DataProviders/ItemStringLoading.lua:4
 NoQueryRefresh = <function> defined @Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua:118
 FilterAuction = <function> defined @Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua:178
 entriesToProcess = <table> {
 }
 presetSort = <table> {
 }
 SetOnResetScrollCallback = <function> defined @Auctionator/Source/Components/DataProviders/DataProvider.lua:138
 SetDirty = <function> defined @Auctionator/Source/Components/DataProviders/DataProvider.lua:130
 processCountPerUpdate = 200
 cachedResults = <table> {
 }
 GetRowTemplate = <function> defined @Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua:304
 AppendEntries = <function> defined @Auctionator/Source/Components/DataProviders/DataProvider.lua:142
 SetPresetSort = <function> defined @Auctionator/Source/Components/DataProviders/DataProvider.lua:54
 ClearSort = <function> defined @Auctionator/Source/Components/DataProviders/DataProvider.lua:60
 CheckForEntriesToProcess = <function> defined @Auctionator/Source/Components/DataProviders/DataProvider.lua:155
 SetOnSearchEndedCallback = <function> defined @Auctionator/Source/Components/DataProviders/DataProvider.lua:122
 cacheUsedCount = 7
 OnLoad = <function> defined @Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua:97
 OnHide = <function> defined @Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua:112
 IsSoldAuction = <function> defined @Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua:173
 GetTableLayout = <function> defined @Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua:296
 OnEvent = <function> defined @Auctionator/Source_Classic/Tabs/DataProviders/Cancelling.lua:145
 SetOnU

```

**Comment by plusmouse** [comment]
Thanks for the bug report. I've put a fix in the latest alpha which will be available momentarily

**Comment by plusmouse** [comment]
Can you try this build and report if the issue is fixed?
[Auctionator-10.1.10-1-g56b560c-wrath.zip](https://github.com/Auctionator/Auctionator/files/11931102/Auctionator-10.1.10-1-g56b560c-wrath.zip)

**Comment by Barov2** [comment]
Sorry, I couldn't test earlier: the problem is fixed in version 10.1.11. Thank you so much !
